add new grounding model TALL for PaddleVideo

huangjun12 opened the pull request:
TALL is a video grounding model, add to PaddleVideo

huangjun12 committed:
add new grounding model TALL for PaddleVideo

huangjun12 committed:
modify configs and add infer data

huangjun12 committed:
delete inference data of tall

SunGaofeng commented on PaddleCV/PaddleVideo/models/tall/tall.py:
```diff
@@ -0,0 +1,169 @@
+            name='train_sentences',
+            shape=[None, sentence_shape],
+            dtype='float32',
+            lod_level=0)
```
检查下是否是变长，如果不是的话，就不用写lod_level了

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/models/tall/tall_net.py:
```diff
@@ -0,0 +1,157 @@
+                                         [1, self.batch_size, 1])
+
+        concat_feature = fluid.layers.concat(
+            [vv_feature, ss_feature], axis=2)  #1,1,2048
```
1，1，2048 应该换成B, B, 2048吧？怕注释误导读者

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/models/tall/tall_net.py:
```diff
@@ -0,0 +1,157 @@
+
+    def net(self, images, sentences):
+        #  fluid.layers.Print(images, message='images', summarize=100)
+        #  fluid.layers.Print(sentences, message='sentences', summarize = 100)
```
删除Print

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/models/tall/tall_net.py:
```diff
@@ -0,0 +1,157 @@
+            param_attr=fluid.param_attr.ParamAttr(name="sim_mat_weights"),
+            bias_attr=False)
+        sim_score_mat = fluid.layers.squeeze(input=sim_score_mat, axes=[0])
+        #  fluid.layers.Print(sim_score_mat, message='sim_score_mat', summarize=100)
```
去掉Print，下同

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/predict.py:
```diff
@@ -147,6 +147,14 @@ def infer(args):
+        elif args.model_name == 'TALL':
+            data_feed_in = [items[:2] for items in data]
+            vinfo = [items[2:] for items in data]
+            video_id = [items[0] for items in vinfo]
```
video_id是item[0]?

从reader部分来看batch_out.append((featmap, sent_vec, start, end, k, t,
                                           movie_clip_sentences,
                                           movie_clip_featmaps, movie_name))
item[0]应该是featmap

huangjun12 replied:
从vinfo中取值，已改成item[6]，对应movie_name

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+import logging
+logger = logging.getLogger(__name__)
+
+import cPickle as pickle
```
python3里面没有cPickle，这样import会出错，请改成下面的形式：
try:
    import cPickle as pickle
except:
    import pickle

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+            movie_clip_names[movie_name].append(k)
+        self.movie_names = list(movie_names_set)
+        logger.info(self.mode.upper() + ':' + str(len(self.movie_names)) +
+                    " movies.")  #
```
去掉 #

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+        # read sliding windows, and match them with the groundtruths to make training samples
+        sliding_clips_tmp = os.listdir(self.sliding_clip_path)  #161396
+        self.clip_sentence_pairs_iou = []
+        if self.mode == 'valid':
```
这地方添加一些注释吧
\# TALL model doesn't take validation during training, it will test after all the training epochs finish.

huangjun12 replied:
ok

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+        if self.mode == 'valid':
+            return
+        if self.mode == 'train':
+            #count = 0
```
去掉注释

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+                    # index_list = list(range(len(self.clip_sentence_pairs_iou)))
+                    # random.shuffle(index_list)
+                    # for i in index_list:
+                    #     clip_sentence_pair = self.clip_sentence_pairs_iou[i]
```
这些注释删掉

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/reader/tall_reader.py:
```diff
@@ -0,0 +1,328 @@
+            elif self.mode == 'test':
+                idx = 0
+                for movie_name in self.movie_names:
+                    idx += 1
```
idx后面好像没有用到？

huangjun12 replied:
Done

SunGaofeng commented on PaddleCV/PaddleVideo/utils/utility.py:
```diff
@@ -52,17 +54,20 @@ def check_cuda(use_cuda, err = \
+        logger.error(err)
+
+
+#         sys.exit(1)
```
加上sys.exit(1)

huangjun12 replied:
Done

huangjun12 committed:
Merge branch 'release/1.6' of https://github.com/PaddlePaddle/models into tall-branch

huangjun12 committed:
delete some redundant notes after gf's review

SunGaofeng commented on PaddleCV/PaddleVideo/models/tall/tall_net.py:
```diff
@@ -0,0 +1,153 @@
+from paddle.fluid import ParamAttr
+import numpy as np
+
+DATATYPE = 'float32'
```
后面好像没有用到？

huangjun12 committed:
refine code of metrics for prediction and evaluation

huangjun12 committed:
fix conficts and refine tall_net code
Merge branch 'release/1.6' of https://github.com/PaddlePaddle/models into tall-branch

huangjun12 committed:
Merge branch 'release/1.6' of https://github.com/PaddlePaddle/models into tall-branch

huangjun12 committed:
delete notes of multi-process reader

SunGaofeng reviewed (approved):
approved
